### DISCUSSION TITLE
JDK Version for Main Project

### DISCUSSION THREAD

**Discussion by tomazfernandes** [spring-projects/spring-kafka#1761] [Q&A] [answered]
Hi, there's one thing I've been working around for some time now but I'd like to clarify... The build.gradle file for the main project specifies java version 1.8, and 11 for tests.

But there are some parts of the code that has features from Java 9, such as @Deprecated(since=...) and now clazz.getPackageName().

What version should I use? I've been craving for some Java 9 goodies such as List.of and Map.of 😄

Thanks!

**Comment by garyrussell** [comment]
We can use `Map.of()` etc in tests; but we have to retain JDK8 compatibility at runtime.

We can use the enhanced `@Deprecated` because it's not used at runtime.

**Reply by tomazfernandes** [reply]
Hmm, that's what I thought before, but it seems now we have in the JsonDeserializer class: 

```
	private void addMappingsToTrusted(Map<String, Class<?>> mappings) {
		mappings.values().forEach(clazz -> {
			doAddTrustedPackages(clazz.getPackageName());
			doAddTrustedPackages(clazz.getPackageName() + ".*");
		});
	}
```
This uses the clazz.getPackageName() that is `* @since 9`, presumably in runtime as well.

The other ones I can just comment out and still compile the code with JDK 8, but if I try to do that with this method some tests fail. Not a big deal, I can work around that, just wanted to be sure on how to proceed.

**Reply by garyrussell** [reply]
Oh; that's a bug. Thanks!

**Reply by tomazfernandes** [reply]
No worries! Thanks!

**Comment by garyrussell** [comment, marked as answer]
#1764
